### PR TITLE
send error packet on io::Error from reader/writer

### LINKED ISSUE #11: connection::Connection should transmit the Error packet
Right now it just uses the Error packet to create an `io::Error`. It'd be best to transmit the courtesy Error packet before returning the `io::Error`. 

Do this for `put` and `get`.

### PR DESCRIPTION
**Pull request by PurpleMyst**

### CODE REVIEW

File: tests/get.rs [hunk 1]
```diff
@@ -37,3 +34,50 @@ fn test_get() {
...
+        Err(io::Error::new(io::ErrorKind::Other, format!("Fake error")))
+    }
+
+    fn flush(&mut self) -> io::Result<()> {
+        Err(io::Error::new(io::ErrorKind::Other, format!("Fake error")))
+    }
+}
+
+// NB: this test just hangs instead of failing before the fix is applied. we
+// could add a timeout here if we wanted it to fail instead.
```
Comment (connorkuehl) [inline tests/get.rs:52]: Sorry, I don't follow. What's meant here by "the fix"?

Comment (PurpleMyst) [reply tests/get.rs:52]: Oh, the fix as in the changes in this commit. If the two tests hang, then there's been a regression. To make them not hang we'd have to introduce timeouts.

Comment (connorkuehl) [reply tests/get.rs:52]: Ah I see now.

I'm not worried about introducing timeouts here. Travis CI will fail a job if it takes too long. Developers can just hit CTRL+C if the test suite is hanging.

**Review by connorkuehl — APPROVED** [review]
Thank you! 🙂

**Pull request merged**